### ISSUE TITLE
More consistent naming of variables in result of foverlaps?

### ISSUE THREAD

**Issue by Henrik-P** [Rdatatable/data.table#1962]
```
x = data.table(start=c(5,31,22,16), end=c(8,50,25,18), val2 = 7:10)
y = data.table(start=c(10, 20, 30), end=c(15, 35, 45), val1 = 1:3)
setkey(y, start, end)
foverlaps(x, y, type="any", which=TRUE) ## return overlap indices
foverlaps(x, y, type="any") ## return overlap join
```

When returning overlap indices, the indices from `x` (`y`) are referred to by using `x` (`y`) as a prefix to `id`. On the other hand, when returning an overlap join, and the join variables have the same name, the variables from `x` have the prefix `i.` (`i.start` and `i.end` in the example). 

I believe it would be more informative to refer to variables from `x` with an `x` also in the second case. Thus, the `i.` notation could be replaced by `x.` when an overlap join is returned.

Furthermore, when overlap indices are returned, although `id` doesn't refer to variables in the original data sets, possibly a `.` may be added to the prefix - `x.id` and `y.id` - to make it more consistent with the "normal" `x.` and `i.` notation in `data.table`.

Just a thought.

**Comment by arunsrinivasan** [comment]
Agreed, but might have to warn in this release and then change in the next one.. Will come back to this.
